### PR TITLE
feat(skills): add ox skills status, and stop judging legacy teams blind

### PR DESCRIPTION
**Pull request by rsnodgrass**
`ox skills status` answers "why isn't my team's skill on my machine?" in one command. Until now that question was effectively unanswerable without reading daemon logs, because **every way it can fail produces the same observation from the repository: nothing there.**

A checkout that hasn't cloned, a sparse set that never materialized the skills root, a repo whose slug matches no `repos:` filter, a skill withheld pending approval, and a team that simply published nothing are all indistinguishable from the outside. The command's job isn't to render state — it's to make those cases distinguishable and end each failing line with the command that fixes it.

## It found a real bug the moment it ran

Pointed at a real team context, the first run reported `agents/ NOT MATERIALIZED`. That was **wrong**, and the reason is a defect shipped in #960:

`unseeableTeamSkills` stat'd only `agents/`, but discovery walks **both** `agents/skills` and the legacy `coworkers/skills`. Every team predating the `agents/` migration was therefore judged permanently blind.

The consequence is quiet and runs the opposite direction from a mass delete: **retirement is suppressed forever.** A skill the team deletes never leaves anyone's machine, and nothing explains why. Fixed here — `anySkillRootOnDisk` keys on `teamdocs.SkillRoots`, the same list discovery uses, so a root can never be walked by one and ignored by the other.

## What the command shows

```
Team Context  SageOx
  path         ~/.local/share/sageox/…/teams/team_…
  checkout     on disk
  agents/      materialized
  last sync    2026-09-17T17:06:44Z

This repo  sageox/ox
  targets      .claude/skills

Team skills  none found
```

Failing states replace the corresponding line and each ends in the fix:

- `checkout  NOT CLONED YET` → *nothing installs until it does; watch with `ox sync --all-teams`*
- `agents/  NOT MATERIALIZED` → *team rules AND skills are invisible here — run `ox doctor`*
- slug from the directory name → *skills with a `repos:` filter cannot match here*
- a withheld skill → named, with the file it bundles, so the reader knows what they'd be approving

`--json` carries the same data plus a `guidance` field holding **one** next action, not a menu — a list of things that might help is a list the caller has to triage, which is the work this command exists to have already done. Guidance lives in the CLI payload rather than a skill body, so Codex and Droid get it too.

## Test Plan

`make lint` — 0 issues. `cmd/ox` and `internal/skillmanager` pass.

| Test | Deliberate break | Result |
|---|---|---|
| `TestLegacyCoworkersRootIsNotBlindness` | restore the `agents/`-only check | `a legacy-rooted team context was reported as unseeable` |

`TestNoSkillRootAtAllIsBlindness` is the counterweight — the real GH #862 shape, where neither root materialized, must still be caught, or the guard would permit a mass delete.

`TestCollectSkillsStatus_DistinguishesTheFailureModes` covers five states and asserts each yields a **different** actionable problem line. That's the product claim; a test that only checked "it printed something" would pass with the command returning one generic message.

## Scope

**`ox skills status` only.** `ox skills publish` writes into a checkout the daemon `git pull --autostash`es every 15 seconds, and that deserves its own review rather than riding along here.

The `docs/reference/` churn is generated — inserting `skills` into the alphabetical ordering shifts `sidebar_position` for everything after it.

Co-authored-by: SageOx <ox@sageox.ai>

---
<a href="https://app.blacksmith.sh/sageox/codesmith/ox/pr/965"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1792257483&installation_model_id=433550&pr_number=965&repository=sageox%2Fox&return_to=https%3A%2F%2Fgithub.com%2Fsageox%2Fox%2Fpull%2F965&signature=02f5ba910d74d031bc138949b113ee4659ffacf108c055117f1651a10859e79e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

## Summary by CodeRabbit

- **New Features**
  - Added `ox skills status` to report repository selection, Team Context state, published skills, installation status, problems, and guidance.
  - Supports human-readable and indented JSON output.
- **Bug Fixes**
  - Improved skill discovery and diagnostics for legacy skill locations, repository-filtered skills, malformed checkouts, and unavailable or incomplete states.
- **Documentation**
  - Added reference documentation for `ox skills` and `ox skills status`.
  - Updated CLI reference navigation to include the new command.

### CODE REVIEW

File: cmd/ox/skills_status.go [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+	TeamSkills  []teamSkillStatus  `json:"team_skills"`
+	Problems    []string           `json:"problems,omitempty"`
+	Guidance    string             `json:"guidance,omitempty"`
+}
+
+type teamContextStatus struct {
+	Name               string `json:"name,omitempty"`
+	Path               string `json:"path,omitempty"`
+	Present            bool   `json:"present"`
+	AgentsMaterialized bool   `json:"agents_materialized"`
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go]: Fixed in 0f568a6a. Renamed to `SkillsMaterialized` / `skills_materialized`, and the human label is now `skill roots`.

You're right that the old name was a lie in the reassuring direction: the check is satisfied by the legacy `coworkers/` root, so a pre-migration team saw `agents/ materialized` for a directory that does not exist in their checkout at all.

File: cmd/ox/skills_status.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+			status.Stale = true
+			out.Problems = append(out.Problems,
+				fmt.Sprintf("the Team Context has not synced in %s — recent edits have not arrived; run `ox sync --all-teams`", roundedAge(sync.LastSync)))
+		}
+	}
+	out.TeamContext = status
+
+	// Discovery tells us what the team publishes; the plan tells us what reached
+	// disk and what was held. Neither alone answers the question.
+	discovered, _ := teamdocs.DiscoverSkills(tc.Path, slug)
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go]: Fixed in 0f568a6a. Both errors now reach `Problems`, and missing-lockfile / missing-root stay valid empty states.

- `skillTargetRoots` wraps `LoadDesired` and returns the error, so an unreadable lockfile reports *"ox could not read this repository's skill lockfile, so what is installed here is unknown"* instead of falling through to the `ox init` recommendation — which was actively wrong advice for an already-initialized repo with a corrupt lockfile.
- Discovery errors report *"ox could not read the team's skills"*. Covered by `TestCollectSkillsStatus_MalformedCheckoutIsReported`, which stages `agents/` as a regular file.
- `skillsStatusGuidance` no longer reaches `Nothing to do.` when anything is pending, unavailable, or unknown. Asserted in both new tests — a diagnostic that says everything is fine while discovery failed is worse than no diagnostic.

File: cmd/ox/skills_status.go [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+				fmt.Sprintf("team skills already installed here are being RETAINED rather than refreshed: %s", reason))
+		}
+	}
+
+	for _, s := range discovered {
+		row := teamSkillStatus{Name: s.Name, AppliesHere: true, State: "installed"}
+		if d, held := withheld[s.Name]; held {
+			row.State = "withheld"
+			row.Detail = d.Reason
+		}
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go]: Fixed in 0f568a6a. This was the real defect in the command and you and Greptile both landed on it.

State now comes from `plan.TeamSkills` plus the file on disk, never from discovery. `installedState` distinguishes five cases explicitly — `installed`, `pending`, `withheld`, `unavailable`, `not applicable` — and a decision with a `Reason` and no `InstalledAs` reports `unavailable`, which `WithheldTeamSkills()` never returned because it only yields approval holds. `planErr` reaches `Problems` and every row degrades to `unknown` rather than claiming anything.

Red-first: defaulting rows to `installed` again fails `TestCollectSkillsStatus_NeverClaimsInstalledFromDiscoveryAlone` with *"a skill that is not on disk was reported installed on the strength of discovery alone."*

File: internal/skillmanager/teamsource.go [hunk 4]
```diff
@@ -258,6 +258,21 @@ func teamRevision(teamPath string) string {
...
+// anySkillRootOnDisk reports whether any directory discovery would walk exists.
+// Keyed on teamdocs.SkillRoots, the same list DiscoverSkills uses, so a root can
+// never be walked by one and ignored by the other.
+func anySkillRootOnDisk(teamPath string) bool {
+	for _, root := range teamdocs.SkillRoots {
+		// The parent is what the sparse set includes ("agents/"), so a materialized
+		// parent with no skills yet is present, not blind.
+		parent := filepath.Dir(filepath.FromSlash(root))
+		if _, err := os.Stat(filepath.Join(teamPath, parent)); err == nil {
+			return true
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply internal/skillmanager/teamsource.go]: Fixed in 0f568a6a. Both checks now require `FileInfo.IsDir()` — `anySkillRootOnDisk` in `internal/skillmanager/teamsource.go` and the status check.

Your reasoning about discovery is the part that made this worth fixing rather than tidying: `os.ReadDir` on `agents/skills` fails when its parent is a regular file, so accepting it would report the checkout healthy while every skill silently failed to load — and it would suppress the blindness guard at the same time, which is the direction that permits deletions.

File: cmd/ox/skills_status.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go]: Fixed in 0f568a6a. Published-but-inapplicable skills now get an explicit `not applicable` row naming the repos they *do* target.

This needed a small split upstream: `teamdocs.DiscoverSkills` applied the `repos:` filter before returning, so the information was gone before the command could see it. `PublishedSkills` now returns everything published to coworkers and `DiscoverSkills` is that plus the filter; `SkillAppliesToRepo` is exported so the command can ask per skill. The audience/visibility/status filters stay inside `PublishedSkills` — a draft or human-only skill is not published anywhere, so surfacing it as "available elsewhere" would be wrong too.

Red-first: skipping the row fails `TestCollectSkillsStatus_RepoFilteredSkillIsVisible` — *"a repos:-filtered skill was invisible, so it looked like the team published nothing."* Which is the point: those two states need opposite actions, author a skill versus widen a list.

File: cmd/ox/skills_status.go [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+		out.TeamSkills = append(out.TeamSkills, row)
+	}
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go]: Fixed in 0f568a6a. Agreed, and fixed as described in the CodeRabbit thread on the same line.

Your framing of the failure is the one I put in the test name: the false success landed *precisely* when the skill had not reached the repository, which is the only moment anyone runs this command. State is now derived from the reconcile decision plus `os.Stat` on the projected path, and a pending create reports `pending` with *"reconcile has not written it yet — run `ox doctor --fix`"*.

File: cmd/ox/skills_status.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go]: Fixed in 0f568a6a. Discovery errors now reach `Problems` rather than rendering as `none found`.

`TestCollectSkillsStatus_MalformedCheckoutIsReported` stages exactly your case — `agents/` as a regular file — and asserts the problem is reported, the root is not counted as materialized, and guidance does not claim everything is current. Telling someone "no skills exist" when the checkout needs repair points them at the wrong action entirely.

File: cmd/ox/skills_status.go [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+// observation from the repository — nothing there. A checkout that has not
+// cloned, a sparse set that never materialized agents/, a repo whose slug does
+// not match a skill's repos: filter, a skill withheld pending approval, and a
+// team that simply published nothing all look identical from the outside.
+//
+// So the command's job is not to render state; it is to make those five cases
+// distinguishable, and to end each failing line with the command that fixes it.
+
+// skillsCmd is the parent for `ox skills …`.
+var skillsCmd = &cobra.Command{
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go:35]: Fixed in 0f568a6a. `skillsCmd.GroupID = "knowledge"`. Verified in built root help — `skills` now sits with `conversation`, `import`, `kb`, `query`, and `recap` instead of under Additional Commands.

**Comment by rsnodgrass** [comment]
@greptileai

File: cmd/ox/skills_status.go [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+	if desiredErr != nil {
+		// A missing lockfile is a valid empty state; anything else means ox cannot
+		// read its own record of what it installed. Suppressing that made the
+		// command recommend `ox init` for a repo that is already initialized and
+		// merely has a corrupt lockfile.
+		out.Problems = append(out.Problems,
+			fmt.Sprintf("ox could not read this repository's skill lockfile, so what is installed here is unknown: %v", desiredErr))
+	}
+	if !selected {
+		out.Problems = append(out.Problems, "this repository has not selected an AI coworker, so no skills are installed — run `ox init`")
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go:158]: Fixed in ff936726 — you caught that my previous fix was only half done. I added `desiredErr` to `Problems` but left the `!selected` branch unconditional, so a corrupt lockfile produced *both* diagnostics and one of them was the wrong action.

Now `if !selected && desiredErr == nil`.

**Red-first**, with the guard removed, the corrupt-lockfile fixture reports:

```
ox could not read this repository's skill lockfile ... invalid character 't' ...
this repository has not selected an AI coworker, so no skills are installed — run `ox init`
```

`TestCollectSkillsStatus_UnreadableLockfileDoesNotRecommendInit` fails with *"a corrupt lockfile was diagnosed as a repository that was never initialized."* Which is the whole problem: `ox init` is the one action that cannot help there.

File: cmd/ox/skills_status.go [hunk 8]
```diff
@@ -0,0 +1,401 @@
...
+	out.TeamContext = status
+
+	// Three different questions, deliberately asked separately:
+	//   what does the team publish        -> PublishedSkills
+	//   which of those are for this repo  -> SkillAppliesToRepo
+	//   which of those are actually here  -> the plan, and the file on disk
+	//
+	// Collapsing any two of them is how a diagnostic reports success at the exact
+	// moment the thing it diagnoses has not happened.
+	published, discoverErr := teamdocs.PublishedSkills(tc.Path)
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go:217]: Fixed in ff936726. Agreed — a successful empty read is a finding, not silence.

`skillsStatusGuidance` now returns, when discovery succeeded, the roots are materialized, and nothing was published:

> Your team has not published any skills yet. Add one under `agents/skills/<name>/SKILL.md` in the Team Context.

And a separate line when there is no Team Context at all, rather than letting both collapse into "current".

Derived from existing fields rather than a new JSON key, so the payload does not grow for a state that is already inferable.

**Red-first**: removing the branch fails `TestCollectSkillsStatus_EmptyPublishedSetIsAnAnswer` with *"an empty-but-healthy team read fell through to generic success guidance: \"Team skills are current. Nothing to do.\""*

Two things that fixture surfaced, both worth keeping: it needs a committed lockfile or the `ox init` problem correctly outranks everything team-related, and it needs a real `origin` remote or the directory-name-slug problem does — so the test asserts `Problems` is empty first, to be sure it is exercising the guidance it claims to.

`make lint` — 0 issues.

**Comment by rsnodgrass** [comment]
@greptileai

File: cmd/ox/skills_status.go [hunk 9]
```diff
@@ -0,0 +1,401 @@
...
+	case d.NeedsApprove:
+		return skillWithheld, d.Reason
+	case d.Name != "" && d.InstalledAs == "":
+		return skillUnavailable, d.Reason
+	}
+	if d.InstalledAs == "" {
+		// No decision at all: the plan did not see it, so it is not ours to claim.
+		return skillUnavailable, "ox did not record a decision for this skill"
+	}
+	for _, root := range targets {
```
[the remark this answers is not in the document]
Comment (rsnodgrass) [reply cmd/ox/skills_status.go:348]: Real bug, and a good find — fixed in ba4b6817.

`installedState` looped over the selected targets and returned `installed` on the first one whose manifest existed. A repo with two roots — `.claude/skills` beside `.agents/skills`, which is what Claude Code plus Codex produces — therefore reported healthy while the second root was missing the skill, holding a stale copy, or had lost a bundled file. Exactly the three cases your T-Rex run drove.

**What changed**

- Every selected target must be complete; the loop now collects `incomplete` and `outdated` rather than returning early, and names the offending root: *"not complete in .agents/skills — run `ox doctor --fix`"*.
- New `outdated` state for present-but-divergent, from `plan.Updates`. Calling that "installed" answers a different question than the one asked.
- Matching is on the skill's **directory prefix**, not `SKILL.md`, so your missing-bundled-file case counts too. The manifest being present says nothing about the rest of the bundle.

**A hole your scenario did not cover, which my own earlier test caught:** with **zero** selected targets the per-target loop never runs, so the collect-then-decide version fell through to `installed` — the most confident possible answer about a repository that has installed nothing. Now returns `pending` with *"this repository has no skills directory selected — run `ox init`"*.

**Red-first**, with any-one-target-is-enough restored:

```
--- Expected
+++ Actual
-outdated
+installed
Messages: a bundle whose contents differ from the team's copy was reported installed
```

`TestInstalledState_EveryTargetMustBeComplete` has four subtests: complete in both, missing from the second, a pending create in the second, and a stale bundled file in the second.

`make lint` — 0 issues.

**Comment by rsnodgrass** [comment]
@greptileai
